packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -356,6 +373,30 @@ export class PendingStateManager implements IDisposable {
         this.isProcessingBatch = false;
     }
 
+    /**
+     * Capture the pending state at this point
+     */
+    public checkpoint() {
```

Comment:
the reason i did some refactoring in my prototype for RuntimeStateHandler was to enabling unit testing for this class. if we go forward with the refactor i'd would like to see some unit tests for the new functionality.